python/ray/serve/tests/unit/test_batching.py

```diff
@@ -319,6 +319,88 @@ async def func(key1, key2):
     assert result == [("hi1", "hi2"), ("hi3", "hi4")]
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize("use_class", [True, False])
+@pytest.mark.parametrize("use_gen", [True, False])
+async def test_cancellation_after_error(use_class, use_gen):
```

Comment:
not totally clear how this test is relevant -- is this how you initially discovered it?

Reply:
let's also add a direct unit test for the behavior that we can execute multiple batches in a row without waiting on the event

Reply:
> is this how you initially discovered it?

Yes, that's right.

> let's also add a direct unit test for the behavior that we can execute multiple batches in a row without waiting on the event

I added `test_batch_wait_queue_exceeds_batch_size_race_condition`.